### ISSUE TITLE
[Feature Request]: Support Model Surgery 

### ISSUE THREAD

**Issue by innat** [keras-team/tf-keras#262]
TensorFlow version (you are using): 2.9.2
Are you willing to contribute it (Yes/No) : 

**Describe the feature and the current behavior/state**.

This request originated from [this](https://stackoverflow.com/q/49492255/9215780) old post. In short, if I've a model initially as follows, I like to able to update the model as such:

- delete layers
- insert layers
- replace layers

For example, in the following `keras_simple_model`:

1. replace `Con2D` layer with the same but without bias.
2. or add `BatchNormalization` layer before first Activation 

```python
def keras_simple_model():
    inputs1 = Input((28, 28, 1))
    x = Conv2D(4, (3, 3), activation=None, padding='same', name='conv1')(inputs1)
    x = Activation('relu')(x)
    x = Conv2D(4, (3, 3), activation=None, padding='same', name='conv2')(x)
    x = Activation('relu')(x)
    x = MaxPooling2D((2, 2), strides=(2, 2), name='pool1')(x)

    x = Conv2D(8, (3, 3), activation=None, padding='same', name='conv3')(x)
    x = Activation('relu')(x)
    x = Conv2D(8, (3, 3), activation=None, padding='same', name='conv4')(x)
    x = Activation('relu')(x)
    x = MaxPooling2D((2, 2), strides=(2, 2), name='pool2')(x)

    x = GlobalAveragePooling2D()(x)
    x = Dense(10, activation=None)(x)
    x = Activation('softmax')(x)

    model = Model(inputs=inputs1, outputs=x)
    return model
```

**Will this change the current api? How?**

I think it is more suited to `tf.keras.utils`.  

**Who will benefit from this feature?**

1. I've faced an [issue](https://github.com/keras-team/keras/issues/17385), where I used augmentaiton layers inside the model. These layers are inactive during the `model.evaluate` and `model.predict` phase. But for test-time-augmentation, I like them to active but that wasn't possible with ease. So, if utility like model surgery is supported, then we can use it inside the callback API to update the target layer to behave differently, `model.predict(tensor, callback=[update])`. With details, 

```python
class TrainingAugmentationLayers(keras.layers.Layer):
    def __init__(self):
        super().__init__()
        self.random_flip = RandomFlip("horizontal")
        self.random_zoom = RandomZoom(0.1, fill_mode='reflect')
        self.random_rotate = RandomRotation(0.1, fill_mode='reflect')
    def call(self, inputs):
        x = self.random_flip(inputs)
        x = self.random_zoom(x)
        x = self.random_rotate(x)
        return x

class PredictAugmentaitonLayers(TrainingAugmentationLayers):
    def call(self, inputs):
        x = self.random_flip(inputs, training=True)
        x = self.random_zoom(x, training=True)
        x = self.random_rotate(x, training=True)
        return x

img = tf.random.normal([20, 512, 512, 3], 0, 1, tf.float32)
y_true = img

inputs = tf.keras.Input(shape=(512, 512, 3))
x = TrainingAugmentationLayers()(inputs)
model = tf.keras.Model(inputs=inputs, outputs=x)
```
```python
class UpdateCallback(keras.callbacks.Callback):
    __init__:
         target_layer_id 
         new_layer

    on_predict_begin:
         keras.utils.update_model(
             self.model, 
             target_layer_id
             new_layer
         )

model.predict(
    tensor, 
    callbakcs=[
        UpdateCallback(
            target_layer_id=0, new_layer=PredictAugmentaitonLayers
        )
    ]
)
```

2. There are some open-source support and quite popular still now. Please check: 
- https://github.com/BenWhetton/keras-surgeon
- https://github.com/leondgarse/keras_cv_attention_models#model-surgery
3. Popular questions/answeres on SO: https://stackoverflow.com/q/49492255/9215780

---

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): Yes (but need some design-level suggestion for all modeling API).
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):
